aten/src/ATen/cpu/vec/sve/sve_helper.h

```diff
@@ -0,0 +1,134 @@
+template<>
+inline uint64_t svcnt<int16_t>() { return svcnth(); }
+template<>
+inline uint64_t svcnt<int32_t>() { return svcntw(); }
+template<>
+inline uint64_t svcnt<int64_t>() { return svcntd(); }
+template<>
+inline uint64_t svcnt<uint8_t>() { return svcntb(); }
+template<>
+inline uint64_t svcnt<uint16_t>() { return svcnth(); }
+template<>
+inline uint64_t svcnt<uint32_t>() { return svcntw(); }
+template<>
+inline uint64_t svcnt<uint64_t>() { return svcntd(); }
+template<>
+inline uint64_t svcnt<float16_t>() { return svcnth(); }
+template<>
+inline uint64_t svcnt<float>() { return svcntw(); }
+template<>
+inline uint64_t svcnt<double>() { return svcntd(); }
+
+// Define the data type of VLS(vector-length specific).
+typedef svbool_t vls_pred_t __attribute__((arm_sve_vector_bits(VECTOR_WIDTH * 8)));
+typedef svint8_t vls_int8_t __attribute__((arm_sve_vector_bits(VECTOR_WIDTH * 8)));
+typedef svint16_t vls_int16_t __attribute__((arm_sve_vector_bits(VECTOR_WIDTH * 8)));
+typedef svint32_t vls_int32_t __attribute__((arm_sve_vector_bits(VECTOR_WIDTH * 8)));
+typedef svint64_t vls_int64_t __attribute__((arm_sve_vector_bits(VECTOR_WIDTH * 8)));
+typedef svuint8_t vls_uint8_t __attribute__((arm_sve_vector_bits(VECTOR_WIDTH * 8)));
+typedef svuint16_t vls_uint16_t __attribute__((arm_sve_vector_bits(VECTOR_WIDTH * 8)));
+typedef svuint32_t vls_uint32_t __attribute__((arm_sve_vector_bits(VECTOR_WIDTH * 8)));
+typedef svuint64_t vls_uint64_t __attribute__((arm_sve_vector_bits(VECTOR_WIDTH * 8)));
+typedef svfloat16_t vls_float16_t __attribute__((arm_sve_vector_bits(VECTOR_WIDTH * 8)));
+typedef svfloat32_t vls_float32_t __attribute__((arm_sve_vector_bits(VECTOR_WIDTH * 8)));
+typedef svfloat64_t vls_float64_t __attribute__((arm_sve_vector_bits(VECTOR_WIDTH * 8)));
+
+// Define all true predicate.
+const vls_pred_t const_ptrue = svptrue_b8();
+
+// Define constant for the value of zero.
+const vls_int8_t const_zero_s8 = svdup_n_s8(0);
+const vls_int16_t const_zero_s16 = svdup_n_s16(0);
+const vls_int32_t const_zero_s32 = svdup_n_s32(0);
+const vls_int64_t const_zero_s64 = svdup_n_s64(0);
+const vls_uint8_t const_zero_u8 = svdup_n_u8(0);
+const vls_uint16_t const_zero_u16 = svdup_n_u16(0);
+const vls_uint32_t const_zero_u32 = svdup_n_u32(0);
+const vls_uint64_t const_zero_u64 = svdup_n_u64(0);
+const vls_float16_t const_zero_f16 = svdup_n_f16(0.f);
+const vls_float32_t const_zero_f32 = svdup_n_f32(0.f);
+const vls_float64_t const_zero_f64 = svdup_n_f64(0.0);
+
+// Define constant for the value of one.
+const vls_int8_t const_one_s8 = svdup_n_s8(1);
+const vls_int16_t const_one_s16 = svdup_n_s16(1);
+const vls_int32_t const_one_s32 = svdup_n_s32(1);
+const vls_int64_t const_one_s64 = svdup_n_s64(1);
+const vls_uint8_t const_one_u8 = svdup_n_u8(1);
+const vls_uint16_t const_one_u16 = svdup_n_u16(1);
+const vls_uint32_t const_one_u32 = svdup_n_u32(1);
+const vls_uint64_t const_one_u64 = svdup_n_u64(1);
+const vls_float16_t const_one_f16 = svdup_n_f16(1.f);
```

Comment:
One can not have something like that in as this is a global constructor that calls an architecture specific instruction

Reply:
Done.
https://github.com/pytorch/pytorch/pull/119571/commits/ceecb567994b8238501fc1386af8ab78235a05a2